azure-devops/azext_devops/dev/repos/policy.py

```diff
@@ -0,0 +1,371 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+# This file a lot of these are intentional because that makes grouping variables easy
+
+from knack.util import CLIError
+from knack.log import get_logger
+from vsts.exceptions import VstsServiceError
+from vsts.policy.v4_0.models.policy_configuration import PolicyConfiguration
+
+from azext_devops.dev.common.services import (get_policy_client, resolve_instance_and_project)
+from azext_devops.dev.common.const import (APPROVER_COUNT_POLICY,
+                                           APPROVER_COUNT_POLICY_ID,
+                                           BUILD_POLICY,
+                                           BUILD_POLICY_ID,
+                                           COMMENT_REQUIREMENTS_POLICY,
+                                           COMMENT_REQUIREMENTS_POLICY_ID,
+                                           MERGE_STRATEGY_POLICY,
+                                           MERGE_STRATEGY_POLICY_ID,
+                                           FILE_SIZE_POLICY,
+                                           FILE_SIZE_POLICY_ID,
+                                           WORKITEM_LINKING_POLICY,
+                                           WORKITEM_LINKING_POLICY_ID,
+                                           REQUIRED_REVIEWER_POLICY,
+                                           REQUIRED_REVIEWER_POLICY_ID)
+from azext_devops.dev.common.identities import resolve_identity_as_id
+
+logger = get_logger(__name__)
+
+
+def list_policy(organization=None, project=None, detect=None):
+    """
+    :param organization: Azure Devops organization URL. Example: https://dev.azure.com/MyOrganizationName/
+    :type organization: str
+    :param project: Name or ID of the project.
+    :type project: str
+    :param detect: Automatically detect organization. Default is "on".
+    :type detect: str
+    :rtype: [PolicyConfiguration]
+    """
+    try:
+        organization, project = resolve_instance_and_project(
+            detect=detect, organization=organization, project=project)
+        policy_client = get_policy_client(organization)
+        return policy_client.get_policy_configurations(project=project)
+    except VstsServiceError as ex:
+        raise CLIError(ex)
+
+
+def get_policy(id, organization=None, project=None, detect=None):  # pylint: disable=redefined-builtin
+    """
+    :param id: ID of the policy.
```

Comment:
This is a mandatory param right? Shouldn't we include required here?

Reply:
knack does that for required params